src/expr.rs

```diff
@@ -189,12 +190,12 @@ impl PyExpr {
     /// Returns the name of this expression as it should appear in a schema. This name
     /// will not include any CAST expressions.
     fn display_name(&self) -> PyResult<String> {
-        Ok(self.expr.display_name()?)
+        Ok(format!("{}", self.expr.schema_name()))
     }
```

Comment:
We should change the `rust` function name to match the upstream change, deprecate `display_name` in the python wrapper and add `schema_name`.

Roughly - the changes in this commit.
https://github.com/Michael-J-Ward/datafusion-python/commit/2cb04b9867cf1e8134c3c8f0afef322c5a774115